File: angular-happykids/src/app/shared/models/childrenItemWarehouse.ts
export interface ChildrenItemWarehouse {
    childrenItemId: number;
    warehouseId: number;
    stockQuantity: number;
    reservedQuantity?: number;
    childrenItem: string;
    warehouse: string;
    city: string;
}

export interface ChildrenItemWarehouseCreateEdit {
    childrenItemId: number;
    warehouseId: number;
    stockQuantity: number;
    reservedQuantity?: number;
}

File: Core/Entities/Warehouses/ChildrenItemWarehouse.cs
namespace Core.Entities.ChildrenItems
{
    public class ChildrenItemWarehouse
    {
        public int ChildrenItemId { get; set; }
        public ChildrenItem ChildrenItem { get; set; }

        public int WarehouseId { get; set; }
        public Warehouse Warehouse { get; set; }

        public int StockQuantity { get; set; }
        public int? ReservedQuantity { get; set; }
    }
}